### ISSUE TITLE
[False-Positive] array_keys() unrecognised type

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1176]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.11 |
| Language level | PHP 7.3                        |

Current behaviour:
---

Here is some example code:

```php
<?php

declare(strict_types=1);

function foo(array $bar): void
{
    foreach (array_keys($bar) as $key) {
        echo $key;
    }
}
```

The following message is displayed:
"Expressions' type was not recognized, please check type hints."

Suggested behaviour:
---

It should be recognised that `array_keys()` returns an `array`.

**Comment by kalessil** [comment]
Thank you for reporting @andreasschroth, I'll fix this.

**Comment by kalessil** [comment]
Checked, but can not reproduce, unfortunately.

**Comment by andreasschroth** [comment]
Please see this screenshot, I'm on Ultimate 2.0.12 now and can still reproduce it:

<img width="1431" alt="screenshot 2019-02-15 at 23 49 25" src="https://user-images.githubusercontent.com/13320551/52871380-83e46580-317c-11e9-89ce-d9396b452090.png">

**Comment by kalessil** [comment]
Will it persist if you create a new project?

**Comment by andreasschroth** [comment]
Funny, you are right.

If I create a new project with just this file, the warning is not displayed. What is the reason for it?

**Comment by kalessil** [comment]
One of the generic PhpStorm problems: caching/indexing data somehow causes failing resolving (due to IDE itself or 3rd party plugins). Invalidating caches and restarting might help.

**Comment by andreasschroth** [comment]
Alright, thanks for the explanation and sorry for opening the issue.

**Comment by kalessil** [comment]
No problems =)
